### PR TITLE
feat: Implement new Ops agent policy module

### PR DESCRIPTION
**Pull request by hsmatulisgoogle**
Create a new module for creating policies to install/uninstall the Ops Agent on GCE VMs

### CODE REVIEW

**Comment by imrannayer** [comment]
@hsmatulisgoogle can u plz follow [this doc](https://github.com/terraform-google-modules/terraform-google-cloud-operations/blob/master/CONTRIBUTING.md)

**Comment by hsmatulisgoogle** [comment]
Fixed the lint issues and believe I followed the doc. Could you please allow the integration tests to run?

File: modules/ops-agent-policy/main.tf [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+                  }
+                  script           = try(exec.value.validate.script, null)
+                  args             = try(exec.value.validate.args, null)
+                  interpreter      = exec.value.validate.interpreter
+                  output_file_path = try(exec.value.validate.outputFilePath, null)
+
+                }
+
+
+                dynamic "enforce" {
```
Comment (g-awmalik) [inline modules/ops-agent-policy/main.tf:274]: nit: whitespace

File: modules/ops-agent-policy/main.tf [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+                    dynamic "file" {
+                      for_each = try(enforce.value.file[*], [])
+                      content {
+                        dynamic "remote" {
+                          for_each = try(file.value.remote[*], [])
+                          content {
+                            uri             = remote.value.uri
+                            sha256_checksum = try(remote.value.sha256Checksum, null)
+                          }
+
```
Comment (g-awmalik) [inline modules/ops-agent-policy/main.tf:286]: nit: whitespace

**Review by g-awmalik — CHANGES_REQUESTED** [review]
LGTM for the most part. Please review the nits and a few comments. I'll approved after you resolve them.

File: modules/ops-agent-policy/main.tf [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+  os_policies {
+    id                            = local.os_policies.id
+    mode                          = local.os_policies.mode
+    allow_no_resource_group_match = local.os_policies.allowNoResourceGroupMatch
+    description                   = "AUTO-GENERATED VALUE BY TERRAFORM, DO NOT EDIT! Enforces Ops Agent installation state."
+
+    dynamic "resource_groups" {
+      for_each = local.os_policies.resourceGroups
+      content {
+        dynamic "inventory_filters" {
```
[the remark this answers is not in the document]
Comment (hsmatulisgoogle) [reply modules/ops-agent-policy/main.tf:62]: They cannot as the inventory_filters in resource_groups comes from a fixed template, and the inventory_filters outside block is user provided. The first one is meant to select the VMs that the policy applies to in general and the second one selects which steps to apply to which VM. Here is an example from the documentation https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/os_config_os_policy_assignment#example-usage---os-config-os-policy-assignment-basic

File: modules/ops-agent-policy/variables.tf [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    var.ops_agent.version == "2.*.*")
+    error_message = "ops_agent.version match one of 'latest', r'2\\.\\d\\.\\d', '2.*.*"
+  }
+}
+
+variable "instance_filter" {
+  description = "Filter to select VMs. Structure is documented below here: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/os_config_os_policy_assignment."
+  type = object({
+    all : optional(bool),
+    exclusion_labels : optional(list(object({
```
[the remark this answers is not in the document]
Comment (hsmatulisgoogle) [reply modules/ops-agent-policy/variables.tf:64]: They cannot. This excludes a VM if it contains all label-value pairs for some element in the list

**Review by g-awmalik — APPROVED** [review]

**Pull request merged**